### DISCUSSION TITLE
Internationalized formatUnits

### DISCUSSION THREAD

**Discussion by Nakochi-crypto** [wevm/viem#1317] [Idea / Feature Request]
`formatUnits` returns a string of an English-style number without digit grouping. However, in some countries `,` is used as a decimal separator and `.` is used as a digit grouping delimiter (e.g. `1.000,1`).

cf. https://en.wikipedia.org/wiki/Decimal_separator

The `Intl.NumberFormat` object enables language-sensitive number formatting.

```
formatUnits(1000100000000000000000n, 18) // => 1000.1

// just an idea
const formatter = new Intl.NumberFormatter('en', { minimumFractionDigits: 2 })
formatUnits(1000100000000000000000n, 18, formatter) // 1,000.10
```

cf. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat

**Comment by tmm** [comment]
Seems like a good idea. Maybe the third positional parameter can be an object of options instead (to make it more future-proof).

```ts
const formatter = new Intl.NumberFormatter('en', { minimumFractionDigits: 2 })
formatUnits(1000100000000000000000n, 18, { formatter })
```
